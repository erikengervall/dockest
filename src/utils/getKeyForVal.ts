import { ObjStrStr } from '../@types'

const { keys } = Object

export default (haystack: ObjStrStr, needle: string): string | undefined =>
  keys(haystack).find(key => haystack[key] === needle)
