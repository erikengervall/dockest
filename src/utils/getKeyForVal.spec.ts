import getKeyForVal from './getKeyForVal'

describe('helpers', () => {
  describe('getKeyForVal', () => {
    it('should return key if value exist on object', () => {
      const haystack = { a: '1', b: '2', c: '3', d: '4', e: '5' }
      const needle = '4'

      const result = getKeyForVal(haystack, needle)

      expect(result).toMatchSnapshot()
    })

    it('should return undefined if value does not exist', () => {
      const haystack = { a: '1', b: '2', c: '3', d: '4', e: '5' }
      const needle = '6'

      const result = getKeyForVal(haystack, needle)

      expect(result).toMatchSnapshot()
    })

    it('should return undefined if value does not exist (2)', () => {
      const haystack = { a: '1', b: '2', c: '3', d: '4', e: '5' }
      const needle = 'c'

      const result = getKeyForVal(haystack, needle)

      expect(result).toMatchSnapshot()
    })
  })
})
