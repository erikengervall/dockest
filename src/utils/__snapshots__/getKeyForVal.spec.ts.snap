// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers getKeyForVal should return key if value exist on object 1`] = `"d"`;

exports[`helpers getKeyForVal should return undefined if value does not exist (2) 1`] = `undefined`;

exports[`helpers getKeyForVal should return undefined if value does not exist 1`] = `undefined`;
